controllers/machine_controller.go

```diff
@@ -83,6 +89,47 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
 
+	mapFn := handler.ToRequestsFunc(
```

Comment:
How about moving this to a function on `r` instead of having it inline here?

Reply:
Which we can then unit test 😄

Reply:
I changed the call to look like this 
```
err = controller.Watch(
		&source.Kind{Type: &clusterv1.Cluster{}},
		&handler.EnqueueRequestsFromMapFunc{
			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request { return r.reconcileRequests(a) }),
		},
		p)
```
where `r.reconcileRequests(a)` is a func that can be unit tested

but the linter fails with 
```
unlambda: replace `func(a handler.MapObject) []reconcile.Request { return r.reconcileRequests(a) }` with `r.reconcileRequests` (gocritic)
```

can someone help with a better idea to inline the func?

Reply:
```go
err = controller.Watch(
		&source.Kind{Type: &clusterv1.Cluster{}},
		&handler.EnqueueRequestsFromMapFunc{
			ToRequests: handler.ToRequestsFunc(r.reconcileRequests(a),
		},
		p)
```
Should work, this would align with how we are adding watches in the Cluster controller: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/cluster_controller.go#L78

Reply:
worked, thanks